control-plane/api-gateway/common/resources.go

```diff
@@ -364,6 +365,25 @@ func (s *ResourceMap) ReferenceCountHTTPRoute(route gwv1beta1.HTTPRoute) {
 	s.httpRouteGateways[consulKey] = set
 }
 
+func (s *ResourceMap) AddHTTPRouteExternalFilter(route gwv1beta1.HTTPRoute, filter client.Object) {
+	key := client.ObjectKeyFromObject(&route)
+	consulKey := NormalizeMeta(s.toConsulReference(api.HTTPRoute, key))
+
+	set, ok := s.httpRouteGateways[consulKey]
+	if !ok {
+		s.ReferenceCountHTTPRoute(route)
```

Comment:
route map needs to be initialized. This shouldn't happen in normal operations, but I did get a panic without this line during testing

Reply:
Since there is an expected order in how you build out these resource dependency maps (i.e. you pull the routes and ref count them first), I'd just see if you can fix up the test instead of doing this here -- I think you'll just need to call the `ReferenceCountHTTPRoute` function prior to calling this function in your test.